catalog/resource_grant.go

```diff
@@ -138,6 +148,9 @@ func ResourceGrant() common.Resource {
 					ConflictsWith: permissions.SliceWithoutString(allFields, field),
 				}
 			}
+
+			m["privileges"].DiffSuppressFunc = permissions.PrivilegesSuppressWhitespaceChange
```

Comment:
@ledbutter  integration test failed, so I looked at this in a bit more details. `DiffSuppressFunc` only works for primitive, i.e. string-like attributes. For lists, we need to look at `CustomizeDiff` instead.

@tanmay-db we probably need to re-add a lightweight version of `CustomizeDiff` to handle this scenario, as it is causing permanent drifts